bin/prettier.js

```diff
@@ -33,15 +41,14 @@ if (!filenames.length && !stdin) {
   process.exit(1);
 }
 
+function formatWithShebang(input) {
+  const programInput = input.split("\n");
```

Comment:
This is a pretty costly operation, especially if the file is big. Can you instead do `indexOf('\n')` and `.slice(0, index)` and `.slice(index)` in order to separate the two parts